From 8d57d0e8c55d0bf78b92c5d24707c6819cd96cb9 Mon Sep 17 00:00:00 2001
From: Niklas Cassel <niklas.cassel@linaro.org>
Date: Tue, 9 Apr 2019 19:03:41 +0200
Subject: [PATCH 15/17] ath10k: sdio: readd start_once

Readd start_once from Erik Stromdahl's high latency patches.
Without this we get DMA transfer errors at boot.
Not sure why this is needed, might be some race at boot.

[    9.447047] mmci-pl18x 121c0000.sdcc: error during DMA transfer!
[    9.447831] ath10k_sdio mmc2:0001:1: failed to read from address 0x874: -110
[    9.453220] ath10k_sdio mmc2:0001:1: failed to read from mbox window data address: -110
[    9.453463] ath10k_sdio mmc2:0001:1: failed to write to address 0x12f00: -110
[    9.459383] ath10k_sdio mmc2:0001:1: failed to read calibration data: -110
[    9.474817] ath10k_sdio mmc2:0001:1: failed to write skb to 0x12f00 asynchronously: -110
[    9.482524] ath10k_sdio mmc2:0001:1: failed to write to address 0x1c700: -110
[    9.489556] ath10k_sdio mmc2:0001:1: failed to write skb to 0x1c700 asynchronously: -110
[    9.498951] ath10k_sdio mmc2:0001:1: failed to write to address 0x1c700: -110
[    9.505090] ath10k_sdio mmc2:0001:1: failed to write skb to 0x1c700 asynchronously: -110
---
 drivers/net/wireless/ath/ath10k/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/wireless/ath/ath10k/core.c b/drivers/net/wireless/ath/ath10k/core.c
index 5d6497210883..08e5fc6f4a6f 100644
--- a/drivers/net/wireless/ath/ath10k/core.c
+++ b/drivers/net/wireless/ath/ath10k/core.c
@@ -512,6 +512,7 @@ static const struct ath10k_hw_params ath10k_hw_params_list[] = {
 		},
 		.hw_ops = &qca988x_ops,
 		.decap_align_bytes = 4,
+		.start_once = true,
 		.num_peers = TARGET_QCA9377_HL_NUM_PEERS,
 		.ast_skid_limit = 0x10,
 		.num_wds_entries = 0x20,
@@ -537,6 +538,7 @@ static const struct ath10k_hw_params ath10k_hw_params_list[] = {
 		.hw_clk = qca6174_clk,
 		.target_cpu_freq = 176000000,
 		.decap_align_bytes = 4,
+		.start_once = true,
 		.num_peers = TARGET_QCA9377_HL_NUM_PEERS,
 		.ast_skid_limit = 0x10,
 		.num_wds_entries = 0x20,
-- 
2.17.1

